tests/unit/test-util/test.expand-prefs.js

```diff
@@ -0,0 +1,68 @@
+import { assert, expect } from 'chai';
+import { describe, it } from 'mocha';
+
+import expandPrefs from '../../../src/util/expand-prefs.js';
+
+describe('utils/expand-prefs', () => {
+  it('expands dot-deliminated preferences into a deep object', () => {
+    const input = {
+      a: 'a',
+      'b.c': 'c',
+      'd.e.f': 'f',
+    };
+    const expected = {
+      a: 'a',
+      b: {
+        c: 'c',
+      },
+      d: {
+        e: {
+          f: 'f',
+        },
+      },
+    };
+    const actual = expandPrefs(input);
+
+    assert.deepEqual(actual, expected);
+  });
+
+  it("doesn't pollute the object prototype", () => {
```

Comment:
The test case here is a specific one that failed the original library that you proposed. Let's also prepend a test for a more obvious test case:

```js
const input = {
  '__proto__.x': 1
};
// Using JSON.parse because an object literal with __proto__
// would be mistaken for an object with a different prototype.
const expected = JSON.parse('{"__proto__":{"x":1}}');
```

Besides the obvious comparison like above, you should also add a test `Object.getPrototypeOf(actual)` and confirm that it is `Object.prototype`. 

And let's also have the same test, but with the following:

```js
const input = JSON.parse('{"__proto__":[]}');
const expected = JSON.parse('{"__proto__":[]}');
```